### ISSUE TITLE
showDialog fires "then" callback when users cancell dialog

### ISSUE THREAD

**Issue by GianFF** [jeduan/cordova-plugin-facebook4#602]
When a user cancells the dialog using the embeded `x` on the facebook pop up (in the pop ups header) the api responds with success and the `then` function is called, but in this case the user hasn't publish anything. 

There are two `x` used for close the pop up: one of them is embeded on the facebook pop up, the other one is embeded on a black circle. I'm referring to the first one.

The `response` is always `" "` so I can not know if the user really publish or not.

I'm using Ionic 2 and testing my app in an Android phone.
This is the code that I'm using to share:
```
  publish(message = "") {
    let publication = {
      method: "share",
      quote: message
    };

    return this.facebook.showDialog(publication)
      .then((/* The user has published. This it not true always, and the response here is always "" */) => { return true; })
      .catch((/* The user close dialog */) => { return false; });
  }
```

Does any one know how to solve this?

**Comment by luckyycool** [comment]
Exactly, and the irony is the response is already blank even if user posts or cancels so there is no way to find if user has published anything

**Comment by noahcooper** [comment]
This plugin is deprecated. Check out `cordova-plugin-facebook-connect` at https://www.npmjs.com/package/cordova-plugin-facebook-connect. In that plugin, the `showDialog` method should correctly fire the success callback only if the user successfully shares, and the error callback if they don't.
